staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -21,39 +21,246 @@ import (
+		if c.Name != containerName {
+			return true
+		}
+		c.VolumeMounts = append(c.VolumeMounts, corev1.VolumeMount{
+			MountPath: "/host",
+			Name:      volumeName,
+		})
+		return false
+	})
+}
+
+// UseHostNamespaces configures the pod to use the host's network, PID, and IPC
+// namespaces.
+func UseHostNamespaces(p *corev1.Pod) {
+	p.Spec.HostNetwork = true
+	p.Spec.HostPID = true
+	p.Spec.HostIPC = true
+}
+
+// ShareProcessNamespace configures all containers in the pod to share the
+// process namespace.
+func ShareProcessNamespace(p *corev1.Pod) {
+	p.Spec.ShareProcessNamespace = pointer.BoolPtr(true)
+}
+
+// ClearSecurityContext clears the security context for the container.
+func ClearSecurityContext(p *corev1.Pod, containerName string, mask podutils.ContainerType) {
+	podutils.VisitContainers(&p.Spec, mask, func(c *corev1.Container, _ podutils.ContainerType) bool {
+		if c.Name != containerName {
+			return true
+		}
+		c.SecurityContext = nil
+		return false
+	})
+}
+
+// DisallowRoot configures the container to run as a non-root user.
+func DisallowRoot(p *corev1.Pod, containerName string, mask podutils.ContainerType) {
+	podutils.VisitContainers(&p.Spec, mask, func(c *corev1.Container, _ podutils.ContainerType) bool {
+		if c.Name != containerName {
+			return true
+		}
+		c.SecurityContext = &corev1.SecurityContext{
+			RunAsNonRoot: pointer.BoolPtr(true),
+		}
+		return false
+	})
+}
+
+// DropCapabilities drops all Capabilities for the container
+func DropCapabilities(p *corev1.Pod, containerName string, mask podutils.ContainerType) {
+	podutils.VisitContainers(&p.Spec, mask, func(c *corev1.Container, _ podutils.ContainerType) bool {
+		if c.Name != containerName {
+			return true
+		}
+		if c.SecurityContext == nil {
+			c.SecurityContext = &corev1.SecurityContext{
+				Capabilities: &corev1.Capabilities{
+					Drop: []corev1.Capability{"ALL"},
+				},
+			}
+			return false
+		}
+		if c.SecurityContext.Capabilities == nil {
+			c.SecurityContext.Capabilities = &corev1.Capabilities{
+				Drop: []corev1.Capability{"ALL"},
+			}
+			return false
+		}
+		c.SecurityContext.Capabilities.Drop = append(c.SecurityContext.Capabilities.Drop, "ALL")
+		return false
+	})
+}
+
+// AllowProcessTracing grants the SYS_PTRACE capability to the container.
+func AllowProcessTracing(p *corev1.Pod, containerName string, mask podutils.ContainerType) {
+	podutils.VisitContainers(&p.Spec, mask, func(c *corev1.Container, _ podutils.ContainerType) bool {
+		if c.Name != containerName {
+			return true
+		}
+		if c.SecurityContext == nil {
+			c.SecurityContext = &corev1.SecurityContext{
+				Capabilities: &corev1.Capabilities{
+					Add: []corev1.Capability{"SYS_PTRACE"},
+				},
+			}
+			return false
+		}
+		if c.SecurityContext.Capabilities == nil {
+			c.SecurityContext.Capabilities = &corev1.Capabilities{
+				Drop: []corev1.Capability{"SYS_PTRACE"},
```

Comment:
Should this be `Capabilities.Add` instead of `Capabilities.Drop`?  Recommend the same as above to avoid this sort of bug:

```
if c.SecurityContext == nil {
  c.SecurityContext = &corev1.SecurityContext{}
}
if c.SecurityContext.Capabilities == nil {
  c.SecurityContext.Capabilities = &corev1.Capabilities{}
}
c.SecurityContext.Capabilities.Add = append(c.SecurityContext.Capabilities.Add, "SYS_PTRACE")
return false
```

Could you add a test case for this?

Reply:
nice catch - fixed and added a test in f604758c84dbf2e9c068455188564a8aaebc7d69